datafusion/common/src/scalar/mod.rs

```diff
@@ -3270,6 +3271,118 @@ impl TryFrom<&DataType> for ScalarValue {
     }
 }
 
+
+impl TryFrom<LogicalType> for ScalarValue {
+    type Error = DataFusionError;
+
+    /// Create a Null instance of ScalarValue for this datatype
+    fn try_from(datatype: LogicalType) -> Result<Self> {
+        (&datatype).try_into()
+    }
+}
+
+impl TryFrom<&LogicalType> for ScalarValue {
+    type Error = DataFusionError;
+
+    /// Create a Null instance of ScalarValue for this datatype
+    fn try_from(data_type: &LogicalType) -> Result<Self> {
```

Comment:
I would tend to agree that ScalarValue should follow LogicalType -- in fact we have had several issues when scalar value types don't quite match (like a logical string encoded as `ScalarValue::String` was compared to a `DictionaryArray` and was treated as a different type

Reply:
It makes sense. I'll add a node to refactor ScalarValues at some point